bigchaindb/common/schema/transaction.yaml

```diff
@@ -105,13 +105,14 @@ definitions:
     description: |
         Description of the asset being transacted. In the case of a ``TRANSFER``
         transaction, this field contains only the ID of asset. In the case
-        of a ``CREATE`` transaction, this field may contain properties:
+        of a ``CREATE`` transaction, this field contains only a user payload
+        associated with the asset.
     additionalProperties: false
-    required:
-    - id
```

Comment:
I'm actually in favour of using the oneOf approach. That way it can match both `{operation:CREATE, asset:{...}}` and `{operation:TRANSFER, asset_id:...}}`. But the more complicated these constraints become, the worse the schema will perform...